src/main/scala/com/datastax/gatling/plugin/model/DseGraphStatements.scala

```diff
@@ -61,60 +68,52 @@ case class GraphFluentStatementFromScalaLambda(lambda: Session => GraphStatement
   *
   * @param sessionKey Place a GraphTraversal in your session with this key name
   */
-case class GraphFluentSessionKey(sessionKey: String) extends DseGraphStatement {
+case class GraphFluentSessionKey(sessionKey: String)
+  extends DseGraphStatement[FluentS, FluentB] {
 
-  def buildFromSession(gatlingSession: Session): Validation[GraphStatement] = {
+  def buildFromSession(gatlingSession: Session): Validation[FluentB] = {
 
     if (!gatlingSession.contains(sessionKey)) {
       throw new DseGraphStatementException(s"Passed sessionKey: {$sessionKey} does not exist in Session.")
     }
 
     Try {
-      DseGraph.statementFromTraversal(gatlingSession(sessionKey).as[GraphTraversal[_, _]])
+      FluentS.builder(gatlingSession(sessionKey).as[GraphTraversal[_, _]])
     } match {
-      case TrySuccess(stmt) => stmt.success
+      case TrySuccess(builder) => builder.success
       case TryFailure(error) => error.getMessage.failure
     }
   }
-
 }
 
 /**
   * Set/Bind Gatling Session key/vals to GraphStatement
   *
-  * @param statement   SimpleGraphStatement
+  * @param builder   SimpleGraphStatementBuilder
   * @param sessionKeys Gatling session param keys mapped to their bind name, to allow name override
   */
-case class GraphBoundStatement(statement: SimpleGraphStatement, sessionKeys: Map[String, String]) extends DseGraphStatement {
+case class GraphBoundStatement(builder: ScriptB, sessionKeys: Map[String, String])
+  extends DseGraphStatement[ScriptS, ScriptB] {
 
   /**
     * Apply the Gatling session params passed to the GraphStatement
     *
     * @param gatlingSession Gatling Session
     * @return
     */
-  def buildFromSession(gatlingSession: Session): Validation[GraphStatement] = {
+
+  def buildFromSession(gatlingSession: Session): Validation[ScriptB] = {
     Try {
-      sessionKeys.foreach((tuple: (String, String)) => setParam(gatlingSession, tuple._1, tuple._2)).success
-      statement
+      sessionKeys foreach {
+        _ match {
+          case (k, v) => builder.setQueryParam(v, gatlingSession(k).as[Object])
+          case _ => throw new RuntimeException()
```

Comment:
I think that we should throw more descriptive error message in such a case

Reply:
That's probably true.  In reality the case statement is there for destructuring of the key-value pair in the map so it's hard to imagine how iterating over a map will produce anything _but_ such a pair... but I agree, if for whatever reason that does happen this probably should be more descriptive.